keps/sig-api-machinery/3903-unknown-version-interoperability-proxy/README.md

```diff
@@ -254,12 +254,36 @@ the number if needed, but upgrades aren't that frequent and few resources are
 changed on releases, so these requests should not be common. We will count them
 with a metric.
 
+There could be a large volume of requests for a specific resource which might result in the identified apiserver being unable to serve the proxied requests. This scenario should not occur too frequently, since resource types which have large request volume should not be added or removed during an upgrade -- that would cause other problems, too.
+
 TODO: security / cert stuff.
 
 ## Design Details
 
-TODO: explanation of how the handler will determine a request is for a resource
-that should be proxied.
+Update the handler stack to detect the condition where proxying of the request to an alternate apiserver is needed.
+```
+func (h *handler) ServeHttp(req *http.Request){
+
+  // find the apiserverid that knows of the request's GVR
+  APIServerIdForResource = StorageVersion.getServerIdsForGVR(req))
+  
+  if (!APIServerIdsForResource) {
+    // no apiserver found that knows about req.resource
+    return (HttpResponse(503))
+  }
+  
+  // get network location of the apiserver along with public key (think of where to make this info available)
+  APIServerLocationInfo = APIService.getNetworkLocation(APIServerIdForResource)
+  
+  // save APIserver network location in request context
+  req = req.WithContext(APIServerLocationInfo)
+  
+  handler.ServeHTTP(w, req)
+}
+```
+
+Concerns:
+* Since we need to query the StorageVersion API to identify which server knows about the requested resource, there will be some latency associated with this lookup. We can make use of a cache to improve performance and at the same time have a refresh interval set for the cache so that it doesn't go out of date.
```

Comment:
We will watch and read from cache. There won't be any serving path latency, just latency seeing new apiservers / forgetting old ones.

Reply:
Updated